src/ImageLoaders/Pef/PefLoaderSegment.cs

```diff
@@ -0,0 +1,130 @@
+#region License
+/* 
+ * Copyright (C) 2018-2021 Stefano Moioli <smxdev4@gmail.com>.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+ */
+#endregion
+using Reko.Core.Memory;
+using Reko.Core.Types;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Reko.ImageLoaders.Pef
+{
+    public class PefLoaderSegment
+    {
+        private readonly PefImageSegment loaderSegment;        
+        private EndianByteImageReader rdr;
```

Comment:
Not too happy with baking in the rdr into what looks like a value class. If this is a loader class, rename it `PefSegmentLoader`, and separate the concerns.